README.md

```diff
@@ -111,6 +111,66 @@ When running on cluster node, you will also need to distribute this keytab, belo
 
 When Spark application is started, it will transparently track the execution plan of submitted SQL/DF transformations, parse the plan and create related entities in Atlas.
 
+Spark models vs Hive models
+====
+
+SAC classifies table related entities with two different kind of models: Spark / Hive.
+
+We decided to determine which models SAC should create based on whether Hive Hook can take care of table related entities or not. For Hive entities, SAC relies on HMS Atlas hook as the source of truth and expects HMS atlas hook to create it.
+
+As SAC cannot check whether the condition is properly made, SAC relies on precondition: if Spark session is connected to the "remote HMS" (via thrift), SAC assumes Hive Hook is already set up in remote HMS and Hive Hook will take care of entities creation.
+
+In other words, SAC assumes table entities are being created in Hive side and just refers these entities via object id if below conditions are true:
+
+* SparkSession.builder.enableHiveSupport is set
+* The value of "hive.metastore.uris" is set to non-empty
+
+For other cases, SAC will create necessary table related entities as Spark models.
+
+One exceptional case is HWC - for HWC source and/or sink, SAC will not create table related entities and always refer to Hive table entities via object id.
+
+Known Limitations (Design decision)
+====
+
+> SAC only supports SQL/DataFrame API (in other words, SAC doesn't support RDD).
+
+SAC relies on query listener to retrieve query and examine the impacts. We may not investigate on supporting RDD for future unless strongly necessary.
+
+> All "inputs" and "outputs" in multiple queries are accumulated into single "spark_process" entity when there're multple queries running in single Spark session.
+
+There're pros and cons regarding which is used for unit of "spark_process". Currently SAC uses "application ID" as unit - this approach could concentrate overall impacts on the single Spark process for who ran for when.
+
+But we also received other voices on this approach - lineage/relationship graph in "spark_process" are too complicated and confused and less meaningful. It's valid voice, we've filed #261 to investigate changing the unit of "spark_process" entity to query. It doesn't mean we will change it soon. It will be address only if we see the clear benefits on changing it.
```

Comment:
Combine the two points above to: 
spark_process maps to an `applicationId` in Spark. This causes lineage/relationship graph in "spark_process" to be complicated and less meaningful. We've filed #261 to investigate changing the unit of "spark_process" entity to query. It doesn't mean we will change it soon. It will be address only if we see the clear benefits on changing it.

Reply:
If we don't describe pros for `applicationId` as id of `spark_process` whereas we describe cons here, it obviously means it has clear benefits on changing and we should just do it. I think we should describe pros and cons of both for rationalization on current status.

Reply:
I don't understand from your message what is the pro of that? Can you please explain?

Reply:
Suppose there's a Spark application which specific user ran from 3 days ago, and the application ran three queries.

If we set ID as query ID for `spark_process`, spark_process will show the impact for specific query but we have to manually collect the impacts from three queries to see the impact for the application. Other trade-off case is query on structured streaming: if the query reads from its checkpoint, query ID doesn't change, which would hide the information when query is re-started.

Other than that, maybe we can track YARN application ID (if yarn scheduler is used) much easier than query id, as the value will be used an ID for both YARN application and Spark application, and the value will be listed in YARN application list, JHS, SHS. For query ID, we may need to dig down log file to figure out which application ran the query, to fix up if something got wrong. Otherwise someone should maintain the information for (application ID, query ID) pairs, which should consider the case for structured streaming query as well. Maybe we could consider adding `application ID` to maintain pair in `spark_process` entity if we want to change the id to query ID.

I'm not saying taking `application ID` as ID of `spark_process` is better. I agree having accumulated sources and sinks on relation graph is in many case less meaningful, but changing the ID would be a breaking change, so I just want to be careful.

Reply:
Nice summary. Looks great to me. I'll reflect the change.